### PR TITLE
feat(tests): add more 7928 test descriptions

### PR DESCRIPTION
**Pull request by nerolation**
## 🗒️ Description

This PR adds three new BAL edge-case tests covering 7702 multi-authorization behavior and the two distinct post-Cancún SELFDESTRUCT outcomes:

1. **7702 Double Authorization Reset:**  
   A single transaction includes two EIP-7702 authorizations for `Alice`—first setting delegation code, then clearing it.  
   BAL must reflect the correct nonce increments (self-funded: 0→3; sponsored: 0→2) and must not include any `code_changes`, since the final code is empty.

2. **SELFDESTRUCT Sending Funds Back to Sender:**  
   A pre-existing contract `C` executes `SELFDESTRUCT(Alice)`.  
   Under EIP-6780, `C` remains intact (no code deletion) and all its balance is transferred to `Alice`.  
   BAL must show `C`’s balance going to zero and `Alice` receiving the value.

3. **SELFDESTRUCT That Burns ETH (Created and Destroyed in Same Tx):**  
   A contract `D` is created and, in the same transaction, executes `SELFDESTRUCT(D)`.  
   Per EIP-6780, the ETH is burned rather than transferred.  
   BAL must show `D`’s balance reduced to zero with no beneficiary receiving the value.

### CODE REVIEW

File: tests/amsterdam/eip7928_block_level_access_lists/test_cases.md [hunk 1]
```diff
@@ -105,3 +105,7 @@
...
+| `test_bal_selfdestruct_burn_balance` | Ensure BAL records ETH burn when a contract created in the same tx selfdestructs to itself | Pre-state: `Alice` sends a CREATE/CREATE2 transaction that deploys contract `D` with an initial endowment of 100 wei. In the same transaction, `D`’s runtime executes and calls `SELFDESTRUCT(D)` (beneficiary is its own address). Under EIP-6780, because `D` is created and selfdestructs in the same transaction and the target equals the caller, the 100 wei is **burned** (removed from total supply). Final post-state: `D` has balance = 0 and is considered destroyed per same-tx semantics. | BAL **MUST** include `D` with `balance_changes` 100→0 (burn) and may include `code_changes` indicating deletion if BAL models code removal; no other account **MUST** have `balance_changes` corresponding to this 100 wei, i.e. no recipient appears. `Alice` **MAY** appear only with `nonce_changes` and gas-related balance changes; the key assertion is that the 100 wei leaves `D` and is not credited to any other account. | 🟡 Planned |
```
Comment (fselmo) [inline tests/amsterdam/eip7928_block_level_access_lists/test_cases.md]: We just merged some more cases and I rebased here. I think that this test case covers this scenario [here](https://github.com/ethereum/execution-specs/blob/eips/amsterdam/eip-7928/tests/amsterdam/eip7928_block_level_access_lists/test_cases.md?plain=1#L97), what do you think?

**Review by fselmo — APPROVED** [review]
lgtm, took me a minute to review and merge some PRs ahead of this one... sorry for the wait here! I just merged those and rebased this one. We may have a duplicate case, can you make sure?

**Comment by nerolation** [comment]
Thanks @fselmo !
Was the duplicated test case removed already, what's the status here?

**Comment by fselmo** [comment]
> Was the duplicated test case removed already, what's the status here?

All I've done is rebase here so that it didn't have conflicts. If you agree that it's testing the same case then we still need to remove it from this list and then merge the other two cases in this PR. What do you think?

**Comment by nerolation** [comment]
looks good!

**Review by fselmo — APPROVED** [review]
